support for stats retrieval

nir0s opened the pull request:
This is a very initial implementation of the stats API.
I wanted to get the ball rolling on this implementation and get your input so that we can perfect this.

Documentation is in place and a stats aggregation feature was added.

nir0s commented:
Some comments.

- Need to add unit tests
- Need to add integration tests
- I have code in place to add a maximum number of stats messages to return in case you don't want to stream indefinitely. Thinking whether this is relevant or not.

shin- commented on docker/client.py:
```diff
@@ -966,11 +966,35 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
+        res = self._get(url, stream=True)
+        self._raise_for_status(res)
+
+        stats_stream = res.iter_lines()
```
Should we use `_stream_helper` here instead?

nir0s replied:
oops. my bad. will do.

shin- commented on docs/api.md:
```diff
@@ -664,6 +664,32 @@ from. Optionally a single string joining container id's with commas
+The Docker API parallel to the `docker stats` command.
+This will stream statistics for a specific container.
+
+`aggregate`, if 0 will return an iterable stream object of messages.
```
I'm unsure what the use case is for this param, can you clarify what your thought process was?

m1keil replied:
I believe this is mechanism for "batching" stats. 
But I think it's a bit problematic since it's blocking and can actually get deadlock if stats stream stopped?

shin- commented:
Thank you for jumpstarting this! I left a couple comments on the code.

> - Need to add unit tests
> - Need to add integration tests

Aye aye, let me know if I can help with those.

> - I have code in place to add a maximum number of stats messages to return in case you don't want to stream indefinitely. Thinking whether this is relevant or not.

Is this something that can't be achieved by:
```python
i = 0
for line in client.stats():
    if i >= MAX:
        break
    ...
```
?

nir0s commented:
I'll see if I get to the tests soon.. if not, i'd appreciate the help.
Regarding the max number of messages.. I was just wondering if it would be useful. Don't wanna add useless code :)

nir0s commented:
Well then.. I added a unittest and am now using stream helper.
I removed aggregation until I have a clearer view of how to implement this.
The reasoning behind it is that you might want to aggregate stats and return only then return them so that you're not blocked by the stream.

Obviously, my previous implementation was blocking. Comes to prove that you shouldn't work long hours :) makes your judgment fuzzy.

shin- commented:
Thank you!
